### PR TITLE
feat: update doctrine-module and doctrine-orm-module

### PR DESCRIPTION
**Pull request by jennifer-choo**
## Description
Updated doctrine-module and doctrine-orm-module dependencies to fix vulnerabilities

Related issue: [BL-15027](https://dvsa.atlassian.net/browse/BL-15027)

## Before submitting (or marking as "ready for review")

- [x] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [x] Have you performed a self-review of the code?
- [ ] Have you have added tests that prove the fix or feature is effective and working?
- [ ] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

**Review by faggc — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -11,16 +11,16 @@
...
-        "doctrine/doctrine-orm-module": "^4.2.1",
-        "laminas/laminas-cache": "^3.1.3",
         "laminas/laminas-db": "^2.9",
         "laminas/laminas-dependency-plugin": "^2",
         "laminas/laminas-eventmanager": "^3.2",
         "laminas/laminas-http": "^2.8",
         "laminas/laminas-log": "^2.10",
         "laminas/laminas-servicemanager": "^3.10",
-        "laminas/laminas-stdlib": "^3.2"
+        "laminas/laminas-stdlib": "^3.2",
```
Comment (caoimheb) [inline composer.json:20]: I'd probably keep these in the order they were in, just more readable alphabetical

File: composer.json [hunk 2]
```diff
@@ -11,16 +11,16 @@
...
+        "doctrine/doctrine-module": "^5.3",
+        "doctrine/doctrine-orm-module": "^5.3",
```
Comment (caoimheb) [inline composer.json:22]: Are we going to test this version bump by just updating mot-logger in MTS first as part of BL-15027? There are some breaking changes flagged in the release docs for this major bump

Comment (jennifer-choo) [reply composer.json:22]: Yeah I'm hitting some version conflicts with mot-logger on the composer.json which is blocking me from continuing on with the repoint ticket. Do you know what they breaking changes are? Currently doing some manual checks on the main mts now and everything seems to be working as expected including the logs however I'll try to do some tests around the breaking changes and will give a heads up to the QAs before test as well

Comment (caoimheb) [reply composer.json]: Some changes / functionality removed is flagged in the release notes - https://github.com/doctrine/DoctrineORMModule/releases - but they might not impact us.

**Review by caoimheb — APPROVED** [review]

**Review by faggc — APPROVED** [review]

**Review by DariuszJerzewski — APPROVED** [review]

File: composer.json (hunk 2 above)
Comment (DariuszJerzewski) [reply composer.json]: It's been regression tested in dev env, and we didn't see any problems. Marking this as resolved :)

**Pull request merged**